plugins/filter/hashids.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Andrew Pantuso (@ajpantuso) <ajpantuso@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.errors import (
+    AnsibleError,
+    AnsibleFilterError,
+    AnsibleFilterTypeError,
+)
+
+from ansible.module_utils.common.text.converters import to_native
+from ansible.module_utils.common.collections import is_sequence
+
+try:
+    from hashids import Hashids
+    HAS_HASHIDS = True
+except ImportError:
+    HAS_HASHIDS = False
+
+
+def initialize_hashids(**kwargs):
+    if not HAS_HASHIDS:
+        raise AnsibleError("The hashids library must be installed in order to use this plugin")
+
+    params = dict((k, v) for k, v in kwargs.items() if v)
+
+    try:
+        return Hashids(**params)
+    except TypeError as e:
+        raise AnsibleFilterError(
+            "The provided parameters %s are invalid: %s" % (
+                ', '.join(["%s=%s" % (k, v) for k, v in params.items()]),
+                to_native(e)
+            )
+        )
+
+
+def hashids_encode(nums, salt=None, alphabet=None, min_length=None):
+    """Generates a youtube-like hash from a sequence of ints
+
+       :nums: Sequence of one or more ints to hash
+       :salt: String to use as salt when hashing
+       :alphabet: String of 16 or more unique characters to produce a hash
+       :min_length: Minimum length of hash produced
+    """
+
+    hashids = initialize_hashids(
+        salt=salt,
+        alphabet=alphabet,
+        min_length=min_length
+    )
+
+    # Handles the case where a single int is not encapsulated in a list or tuple.
+    # User convenience seems prefferable to strict typing in this case
+    # Also avoids obfuscated error messages related to single invalid inputs
+    if not is_sequence(nums):
+        nums = [nums]
+
+    try:
+        hashid = hashids.encode(*nums)
+    except TypeError as e:
+        raise AnsibleFilterTypeError(
+            "Data to encode must by a tuple or list of ints: %s" % to_native(e)
+        )
+
+    return hashid
+
+
+def hashids_decode(hashid, salt=None, alphabet=None, min_length=None):
+    """Decodes a youtube-like hash to a sequence of ints
+
+       :hashid: Hash string to decode
+       :salt: String to use as salt when hashing
+       :alphabet: String of 16 or more unique characters to produce a hash
+       :min_length: Minimum length of hash produced
+    """
+
+    hashids = initialize_hashids(
+        salt=salt,
+        alphabet=alphabet,
+        min_length=min_length
+    )
+    nums = hashids.decode(hashid)
```

Comment:
I think it's better to convert the returned tuple to a list instead, since tuples are something that Ansible users would not expect (Ansible users usually only know JSON types, so they would expect a list). What do you think?

Reply:
Yes, returning a tuple of arbitrary size doesn't make sense anyway so there's no downside to adding this convenience. Will push an update soon.

Reply:
In Python code it does make sense (tuples are immutable sequences, while lists are mutable), but this isn't used in real Python code :)

Reply:
I didn't mean in the pythonic or even mathematical sense in this case. To me the unpacking syntax for tuples in python makes it more important to be consistent about the number of return values. Especially since Go has promoted the `value, err := ...` pattern.

Reply:
The unpacking syntax also works for lists. I think it generally works for sequences, which for example includes strings, as you can check with:
```.py
x, y, z = [1, 2, 3]
x, y, z = '123'
```